gateway/app/main/events/fl_events.py

```diff
@@ -0,0 +1,153 @@
+
+        # Create a new FL Process
+        fl_process = processes.create_process(
+            model=serialized_model,
+            client_plans=serialized_client_plans,
+            client_protocols=serialized_client_protocols,
+            server_averaging_plan=serialized_avg_plan,
+            client_config=client_config,
+            server_config=server_config,
+        )
+
+    except Exception as e:  # Retrieve exception messages such as missing JSON fields.
+        response[RESPONSE_MSG.ERROR] = str(e)
+
+    return json.dumps(response)
+
+
+def authenticate(message: dict, socket) -> str:
+    """ New workers should receive a unique worker ID after authenticate on PyGrid platform.
+        
+        Args:
+            message : Message body sended by some client.
+            socket: Socket descriptor.
+        Returns:
+            response : String response to the client
+    """
+    response = {}
+
+    # Create a new worker instance and bind it with the socket connection.
+    try:
+        # Create new worker id
+        worker_id = str(uuid.uuid4())
+
+        # Create a link between worker id and socket descriptor
+        handler.new_connection(worker_id, socket)
+
+        # TODO:
+        # Create a new User/Worker instance to map and manage this worker.
+
+        response[MSG_FIELD.WORKER_ID] = worker_id
+
+    except Exception as e:  # Retrieve exception messages such as missing JSON fields.
+        response[RESPONSE_MSG.ERROR] = str(e)
+
+    return json.dumps(response)
+
+
+def cycle_request(message: dict, socket) -> str:
+    """This event is where the worker is attempting to join an active federated learning cycle. 
+        
+        Args:
+            message : Message body sended by some client.
+            socket: Socket descriptor.
+        Returns:
+            response : String response to the client
+    """
+    data = message[MSG_FIELD.DATA]
+    response = {}
+
+    try:
+        # Retrieve JSON values
+        worker_id = data.get(MSG_FIELD.WORKER_ID, None)
+        model_id = data.get(MSG_FIELD.MODEL, None)
+        version = data.get(CYCLE.VERSION, None)
+        ping = data.get(CYCLE.PING, None)
+        download = data.get(CYCLE.DOWNLOAD, None)
+        upload = data.get(CYCLE.UPLOAD, None)
+
+        # TODO:
+        # Search/Identify worker id
+        # Search/Identify model id
+        # Check worker health (ping/download/upload rate)
+        # If available,  add this worker on fd cycle
+
+        ### MOCKUP ###
+
+        # Build response
+        accepted = True
+        if accepted:
+            response[CYCLE.STATUS] = "accepted"
+            response[CYCLE.KEY] = "LONG HASH KEY"
+            response[CYCLE.PLANS] = {}
+            response[CYCLE.PROTOCOLS] = {}
+            response[CYCLE.CLIENT_CONFIG] = {}
+            response[MSG_FIELD.MODEL] = model_id
+        else:
+            response[CYCLE.STATUS] = "rejected"
+            response[CYCLE.TIMEOUT] = 2700
```

Comment:
In theory, the worker should only re-request to join the NEXT cycle. In the real world, this value should be "the number of seconds remaining in the current cycle + 1".

Reply:
@IonesioJunior ?

Reply:
Ohh, sorry I didn't see this comment. I'll fix it! :)